src/app/InteractionModelEngine.h

```diff
@@ -0,0 +1,182 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines objects for a CHIP Interaction Data model Engine which handle unsolicitied IM message, and
+ *      manage different kinds of IM client and handlers.
+ *
+ */
+
+#pragma once
+
+#ifndef _CHIP_INTERACTION_MODEL_ENGINE_H
+#define _CHIP_INTERACTION_MODEL_ENGINE_H
+
+#include <app/MessageDef.h>
+#include <core/CHIPCore.h>
+#include <map>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+#include <app/Command.h>
+#include <app/CommandHandler.h>
+#include <app/CommandSender.h>
+
+#define CHIP_MAX_NUM_COMMAND_HANDLER_OBJECTS 1
+#define CHIP_MAX_NUM_COMMAND_SENDER_OBJECTS 1
+
+namespace chip {
+namespace app {
+
+typedef void (*CommandCbFunct)(chip::TLV::TLVReader & aReader, Command * apCommandObj);
```

Comment:
nit: Prefer the more readable syntax

`using CommandCallback = void(chip::TLV::TLVReader & aReader, Command * apCommandObj);`

and one below.

Reply:
After applying alias-declaration, I am seeing this error,
/usr/include/c++/10/bits/stl_pair.h:218:11: error: data member ‘std::pair<const chip::app::InteractionModelEngine::HandlerKey, void(chip::TLV::TLVReader&, chip::app::Command*)>::second’ invalidly declared function type
  218 |       _T2 second;                ///< The second membe

I just create a new ticket remove typedef declaration

Reply:
https://github.com/project-chip/connectedhomeip/issues/4276